src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -153,7 +153,7 @@ export class BrowserRenderer {
         return this.insertFrameRange(batch, componentId, parent, childIndex, frames, frameIndex + 1, frameIndex + frameReader.subtreeLength(frame));
       case FrameType.elementReferenceCapture:
         if (parent instanceof Element) {
-          applyCaptureIdToElement(parent, frameReader.elementReferenceCaptureId(frame));
+          applyCaptureIdToElement(parent, frameReader.elementReferenceCaptureId(frame)!);
```

Comment:
Educational for me... what does the `!` mean in this context?

Reply:
@rynowak '!' means that you're sure the object will have some value (a.k.a, you're sure it's not going to be NULL) sometimes the compiler can complain if it's a nullable type. But if you're sure it's not going to be NULL in this context, you can overwrite it using '!'

Reply:
cool, thanks!

Reply:
Yep, TypeScript treats types as non-nullable by default in `strict` mode, which have enabled.

In this case, `elementReferenceCaptureId` is declared as returning `string | null`, whereas `applyCaptureIdToElement` requires a `string`. Since we know we only go down this code path for the `FrameType.elementReferenceCapture` case, we know the `elementReferenceCaptureId` will be nonnull.